tokio/src/runtime/context.rs

```diff
@@ -24,11 +33,27 @@ pub(crate) fn current() -> Handle {
     }
 }
 
+pub(crate) fn set_current_task_id(id: Option<Id>) -> Option<Id> {
+    CONTEXT.with(|ctxt| ctxt.task_id.replace(id))
+}
+
+#[track_caller]
+pub(crate) fn current_task_id() -> Id {
+    match CONTEXT.try_with(|ctxt| ctxt.task_id.get()) {
+        Ok(Some(id)) => id,
+        Ok(None) => panic!("tried to get current task id from outside of task"),
+        Err(e) => panic!(
+            "tried to get current task id from outside of task failed with error {:?}",
+            e
+        ),
```

Comment:
There isn't any reason to print the thread-local error here. The reason is the same either way - we are not inside the task.
```suggestion
        _ => panic!("tried to get current task id from outside of task"),
```